sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -51,6 +51,93 @@ APIs for managing module identities, module twins, and querying modules
+
+    /// <summary>
+    /// List a set of device modules.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>A pageable set of device modules.</returns>
+    public virtual async AsyncPageable<ModuleIdentity> GetIdentitiesAsync(string deviceId, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Update a module.
+    /// </summary>
+    /// <param name="moduleIdentity">The module to update.</param>
+    /// <param name="ifMatch">A string representing a weak ETag for this module, as per RFC7232. The update operation is performed
+    /// only if this ETag matches the value maintained by the server, indicating that the module has not been modified since it was last retrieved.
+    /// The current ETag can be retrieved from the module identity last retrieved from the service. To force an unconditional update, set If-Match to the wildcard character (*).</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The created or updated device module.</returns>
+    public virtual async Task<Response<ModuleIdentity>> UpdateIdentityAsync(ModuleIdentity moduleIdentity, string ifMatch = null, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Delete a single module.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device that contains the module.</param>
+    /// <param name="moduleId">The unique identifier of the module to get.</param>
+    /// <param name="ifMatch">A string representing a weak ETag for this module, as per RFC7232. The update operation is performed
+    /// only if this ETag matches the value maintained by the server, indicating that the module has not been modified since it was last retrieved.
+    /// The current ETag can be retrieved from the module identity last retrieved from the service. To force an unconditional update, set If-Match to the wildcard character (*).</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The http response.</returns>
+    public virtual async Task<Response> DeleteIdentityAsync(string deviceId, string moduleId, string ifMatch = null, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Gets the module twin.
+    /// </summary>
+    /// <param name="deviceId">The unique identifier of the device.</param>
+    /// <param name="moduleId">The unique identifier of the device module.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The retrieved module twin.</returns>
+    public virtual async Task<Response<TwinData>> GetTwinAsync(string deviceId, string moduleId, CancellationToken cancellationToken);
```

Comment:
Did we think of having ModuleTwinData and DeviceTwinData extending from TwinData? Maybe a good time for you to evaluate.

Reply:
I think they are the same thing, that may be worth doing if they diverged , we can discuss this in the meeting.